portal-web/test/functional/com/liferay/portalweb/tests/enduser/contentperformance/ContentPerformance.testcase

```diff
@@ -89,21 +89,90 @@ definition {
 	}
 
 	@description = "LPS-105216: Validate that Free trial and dismiss buttons ARE NOT displayed when AC is enabled"
-	@ignore = "true"
 	@priority = "3"
 	test FreeTrialButtonNotPresent {
+		task ("Assert if element is not present") {
+			ContentPerformance.contentPerformanceWithoutMessage(message = "Start Free Trial");
 
-		// TODO LPS-139217 FreeTrialButtonNotPresent pending implementation
-
+			ContentPerformance.contentPerformanceWithoutMessage(message = "Do not show me this again.");
+		}
 	}
 
 	@description = "LPS-105216: Validate that Free trial and dismiss buttons are displayed when AC is disabled"
-	@ignore = "true"
 	@priority = "5"
 	test FreeTrialButtonPresent {
 
-		// TODO LPS-139217 FreeTrialButtonPresent pending implementation
+		task ("Open Content Performance") {
+
+			var testPortalInstance = PropsUtil.get("test.portal.instance");
+
+			Navigator.openURL();
+
+			AnalyticsCloud.teardownAnalyticsCloud();
+
+			JSONGroup.deleteGroupByName(groupName = "Test Site Name");
+
+			if ("${testPortalInstance}" == "true") {
+			PortalInstances.tearDownCP();
+			}
+
+			Navigator.openURL();
+		}
+
+		task ("Go to DXP portal, login and create site") {
+			TestCase.setUpPortalInstance();
+
+			User.firstLoginPG();
+
+			Navigator.openURL();
+
+			JSONGroup.addGroup(groupName = "Test Site Name");
+		}
+
+		task ("Go to DXP portal, login and assert if element is present") {
+			ContentPerformance.openContentPerformance();
+
+			ContentPerformance.contentPerformanceMessage(message = "Start Free Trial");
+			
+			ContentPerformance.contentPerformanceMessage(message = "Do not show me this again.");
+		}
+
+		task ("Open Analytics Cloud") {
+			var analyticsCloudURL = PropsUtil.get("analytics.cloud.url");
+
+			Navigator.openURL(baseURL = "${analyticsCloudURL}");
+		}
+
+		task ("Take token from Analytics Cloud") {
+			AnalyticsCloud.loginAnalyticsCloud(
+				emailAddress = "test@liferay.com",
+				password = "test");
+
+			AnalyticsCloud.takeTokenFromAnalyticsCloud();
+		}
+
+		task ("Go to DXP portal, login and create site") {
+			TestCase.setUpPortalInstance();
+
+			User.firstLoginPG();
+
+			Navigator.openURL();
+
+			JSONGroup.addGroup(groupName = "Test Site Name");
+		}
```

Comment:
There's no need to connect DXP with AC, the goal of this test is just to assert the messages when DXP is not connected to AC.

Reply:
I thought a lot about this test, how best to do it. but all other tests connect in ac, so it's interesting that the synchronization between ac and dxp happens in set up. this test is an exception and it was necessary to make this move